File: java/src/main/java/deltix/luminary/LiteralList.java
package deltix.luminary;

import java.util.ArrayList;
import java.util.List;

public final class LiteralList extends Literal {
    private final List<Literal> value;

    public LiteralList() {
        super(LiteralKind.LIST);
        value = new ArrayList<>();
    }

    public List<Literal> getValue() {
        return value;
    }

    @Override
    public String toString() {
        if (getValue().size() == 0)
            return "{}";

        final StringBuilder builder = new StringBuilder("{ ").append(value.get(0));
        for (int i = 1; i < value.size(); i += 1)
            builder.append(", ").append(value.get(i));
        return builder.append(" }").toString();
    }
}

File: csharp/main/LiteralList.cs
using System;
using System.Collections.Generic;

namespace Deltix.Luminary
{
	public class LiteralList : Literal, ILiteral<List<Literal>>
	{
		public LiteralList()
			: base(LiteralKind.List)
		{
			Value = new List<Literal>();
		}

		public List<Literal> Value { get; }

		public override String ToString()
		{
			return $"{{{String.Join(", ", Value)}}}";
		}
	}
}
